### ISSUE TITLE
Autofix on `FLY002` produces a syntax error

### ISSUE THREAD

**Issue by Arborym** [charliermarsh/ruff#4395]
Code Snippet:
```py
a = "bar"
b = "baz"

c = "x".join([a, "foo", b])
print(c)
```
After Autofix:
```py
a = "bar"
b = "baz"

c = "f"{a}xfoox{b}"
print(c)
```
It looks like the double quote marks are misplaced and create a syntax error

**Comment by charliermarsh** [comment]
I'm unable to reproduce with that snippet -- is this the entire file?

**Comment by Arborym** [comment]
Yes, it's the entire file. Here's a recording of what I did:
[screen-capture.webm](https://github.com/charliermarsh/ruff/assets/105245560/24ba75f5-43b8-4e90-93e5-a3123ac0f2d5)

**Comment by Arborym** [comment]
Thank you! 😃

**Comment by charliermarsh** [comment]
Fixed in v0.0.267 with the latest VS Code release.
